cmd/containerd-shim-runhcs-v1/delete.go

```diff
@@ -44,9 +65,16 @@ The delete command will be executed in the container's bundle as its cwd.
 		// log those messages (if any) on stderr so that it shows up in containerd's log.
 		// This should be done as the first thing so that we don't miss any panic logs even if
 		// something goes wrong during delete op.
-		logs, err := ioutil.ReadFile(filepath.Join(bundleFlag, "panic.log"))
-		if err == nil && len(logs) > 0 {
-			logrus.WithField("log", string(logs)).Error("found shim panic logs during delete")
+		// The file can be very large so read only first 1MB of data.
+		readLimit := int64(1024 * 1024) // 1MB
+		logBytes, err := limitedRead(filepath.Join(bundleFlag, "panic.log"), readLimit)
+		if err == nil {
+			if int64(len(logBytes)) == readLimit {
+				logrus.Warnf("shim panic log file %s is larger than 1MB, logging only first 1MB", filepath.Join(bundleFlag, "panic.log"))
+			}
+			logrus.WithField("log", string(logBytes)).Warn("found shim panic logs during delete")
+		} else if !errors.As(err, &os.ErrNotExist) {
```

Comment:
`errors.Is` might make more sense here because we don't care about using the second argument afterwards, but rather we just want to see if err was wrapped with `os.ErrNotExist`